src/main/java/duchess/util/Parser.java

```diff
@@ -0,0 +1,206 @@
+                try {
+                    String[] eventDetails = getEventDetails(input);
+                    Task t = new Event(eventDetails[0], eventDetails[1], eventDetails[2]);
+
+                    return tasks.createTask(t);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    response += LINE_BREAK;
+                    response += "\nIncorrect input. ";
+                    response += "\nTo create an 'Event': \nevent <description> /from <from> /to <to>";
+                    response += LINE_BREAK;
+                    e.printStackTrace(System.err);
+
+                    return response;
+                }
+            } else if (input.substring(0, 6).toUpperCase().contains("DELETE")) {
+                try {
+                    int itemIndex = Character.getNumericValue(input.charAt(7));
+
+                    return tasks.deleteTask(itemIndex);
+                } catch (IndexOutOfBoundsException e) {
+                    response += LINE_BREAK;
+                    response += "\nPlease specify a valid task number.";
+                    response += LINE_BREAK;
+                    e.printStackTrace(System.err);
+
+                    return response;
+                }
+            } else if (input.substring(0, 6).toUpperCase().contains("UNMARK")) {
+                try {
+                    int itemIndex = Character.getNumericValue(input.charAt(7));
+
+                    return tasks.unmarkTask(itemIndex);
+                } catch (IndexOutOfBoundsException e) {
+                    response += LINE_BREAK;
+                    response += "\nOOPS!!! Please specify a valid task number.";
+                    response += LINE_BREAK;
+                    e.printStackTrace(System.err);
+
+                    return response;
+                }
+            } else if (input.substring(0, 8).toUpperCase().contains("DEADLINE")) {
+                try {
+                    String[] deadlineDetails = getDeadlineDetails(input);
+                    Task t = new Deadline(deadlineDetails[0], deadlineDetails[1]);
+
+                    return tasks.createTask(t);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    response += LINE_BREAK;
+                    response += "\nIncorrect input. ";
+                    response += "\nTo create a 'Deadline': \ndeadline <description> /by <by>";
+                    response += LINE_BREAK;
+                    e.printStackTrace(System.err);
+
+                    return response;
+                }
+            }
+        } catch (StringIndexOutOfBoundsException e) {
+            response += LINE_BREAK;
+            response += "\nOOPS!!! I'm sorry, but I don't know what that means.";
+            response += "\nEnter 'MENU' to view available options.";
+            response += "\n" + LINE_BREAK;
+
+            throw new DuchessException(response);
+        }
+
+        return response;

```

Comment:
Just a tiny suggestion to give the "response" variable a more descriptive name, such as "taskListResponse" or "deleteTaskResponse," to improve readability. Otherwise, your code is well-organized. Good job!